File: ios/Exponent/Kernel/Views/Loading/EXAppLoadingProgressWindowViewController.m
#import "EXAppLoadingProgressWindowViewController.h"
#import "EXKernel.h"
#import "EXAppViewController.h"

@implementation EXAppLoadingProgressWindowViewController

- (UIInterfaceOrientationMask)supportedInterfaceOrientations
{
  const UIInterfaceOrientationMask visibleAppSupportedInterfaceOrientations =
    [EXKernel sharedInstance]
      .visibleApp
      .viewController
      .supportedInterfaceOrientations;
  return visibleAppSupportedInterfaceOrientations;
}

@end

File: ios/Exponent/Kernel/Views/Loading/EXAppLoadingProgressWindowViewController.swift
import UIKit

@objc
class EXAppLoadingProgressWindowViewController: UIViewController {
  override var supportedInterfaceOrientations: UIInterfaceOrientationMask {
    let visibleAppSupportedInterfaceOrientations = EXKernel
      .sharedInstance()
      .visibleApp
      .viewController
      .supportedInterfaceOrientations
    return visibleAppSupportedInterfaceOrientations
  }
}
